types/bull/index.d.ts

```diff
@@ -55,7 +55,7 @@ declare namespace Bull {
     /**
      * Options passed directly to the `ioredis` constructor
      */
-    redis?: Redis.RedisOptions | string | undefined;
```

Comment:
If these are truly options directly passed to ioredis (what the docstring says), you can pass a string (URI).

https://github.com/luin/ioredis#connect-to-redis

If this is incorrect, please update the doc string with that it does.

Reply:
it used to be, but now bull clones the object before passing it to redis and overwrites `connectionName`

docs on bull about this are vague, there is only a comment in an example saying to use the object form ([here](https://github.com/OptimalBits/bull#basic-usage))

Reply:
I don't think removing string here is accurate, Bull fully supports passing a string as optio. See [here](https://github.com/OptimalBits/bull/blob/c2194fead19e8d7a1159672e2f21de2a29d39990/lib/queue.js#L92-L100)

Reply:
Oh. I think I was mistaken and you're right @m-ronchi. That was a different part of the code. When using an object, the redis key should not be a string. See [here](https://github.com/OptimalBits/bull/blob/c2194fead19e8d7a1159672e2f21de2a29d39990/lib/queue.js#L123-L130)